File: JS Advanced/15. Modules, Babel, CommonJS Exercises/01. People Modules/employee.js
class Employee{
    constructor(name, age){
        if(new.target == Employee){
            throw new Error("Cannot make instance of abstract class Employee.");
        }
        this.name = name;
        this.age = age;
        this.salary = 0;
        this.tasks = [];
    }

    work(){
        let currentTask = this.tasks.shift();
        console.log(`${this.name} ` + currentTask);
        this.tasks.push(currentTask);
    }

    getSalary(){
        return this.salary;
    }

    collectSalary(){
        console.log(`${this.name} received ${this.getSalary()} this month.`)
    }
}

module.exports = Employee;

File: C# OOP Advanced/06. Object Communication and Events Exercises/04. Work Force/Models/Employee.cs
using System;
using System.Collections.Generic;
using System.Text;

public abstract class Employee
{
    protected Employee(string name, int hoursPerWeek)
    {
        this.Name = name;
        this.HoursPerWeek = hoursPerWeek;
    }

    public int HoursPerWeek { get; private set; }

    public string Name { get; private set; }
}